src/ol/expr/gpu.js

```diff
@@ -224,6 +226,18 @@ const compilers = {
     const prefix = context.inFragmentShader ? 'v_prop_' : 'a_prop_';
     return prefix + propName;
   },
+  [Ops.Id]: (context) => {
+    const isExisting = FEATURE_ID_PROPERTY_NAME in context.properties;
+    if (!isExisting) {
+      context.properties[FEATURE_ID_PROPERTY_NAME] = {
+        name: FEATURE_ID_PROPERTY_NAME,
+        type: NumberType | StringType,
+        evaluator: (feature) => feature.getId() ?? null,
+      };
+    }
+    const prefix = context.inFragmentShader ? 'v_prop_' : 'a_prop_';
+    return prefix + FEATURE_ID_PROPERTY_NAME;
```

Comment:
The `prop` prefix only needs to be used when we are declaring variables using user-supplied names. This includes feature properties and style variables. The `prop` prefix does not need to be used for names that we control or know (like "featureId”).

I know this is an existing problem (prop is used where it doesn’t need to be in the heatmap layer and elsewhere), and I understand if you don’t want to address it here. Just another thing to address at some point.

Reply:
Ok, didn't realize that, thanks for the heads-up! I'll address it in this PR